hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/token/CryptoPreTransactionHandler.java

```diff
@@ -20,15 +20,278 @@
+     * required signing keys.
+     *
+     * @param txn a transaction with a {@link
+     *     com.hederahashgraph.api.proto.java.CryptoDeleteTransactionBody}
+     * @return the metadata for the account deletion
+     */
+    TransactionMetadata preHandleCryptoDelete(TransactionBody txn);
+
+    /**
+     * Pre-handles a {@link
+     * com.hederahashgraph.api.proto.java.HederaFunctionality#CryptoApproveAllowance} transaction,
+     * returning the metadata required to, at minimum, validate the signatures of all required
+     * signing keys.
+     *
+     * @param txn a transaction with a {@link
+     *     com.hederahashgraph.api.proto.java.CryptoApproveAllowanceTransactionBody}
+     * @return the metadata for the allowance approvals
+     */
+    TransactionMetadata preHandleApproveAllowances(TransactionBody txn);
+
+    /**
+     * Pre-handles a {@link
+     * com.hederahashgraph.api.proto.java.HederaFunctionality#CryptoDeleteAllowance} transaction,
+     * returning the metadata required to, at minimum, validate the signatures of all required
+     * signing keys.
+     *
+     * @param txn a transaction with a {@link
+     *     com.hederahashgraph.api.proto.java.CryptoDeleteAllowanceTransactionBody}
+     * @return the metadata for the allowance revocations
+     */
+    TransactionMetadata preHandleDeleteAllowances(TransactionBody txn);
+
+    /**
+     * Pre-handles a {@link
+     * com.hederahashgraph.api.proto.java.HederaFunctionality#CryptoAddLiveHash} transaction,
+     * returning the metadata required to, at minimum, validate the signatures of all required
+     * signing keys.
+     *
+     * @param txn a transaction with a {@link
+     *     com.hederahashgraph.api.proto.java.CryptoAddLiveHashTransactionBody}
+     * @return the metadata for the live hash addition
+     */
+    TransactionMetadata preHandleAddLiveHash(TransactionBody txn);
+
+    /**
+     * Pre-handles a {@link
+     * com.hederahashgraph.api.proto.java.HederaFunctionality#CryptoDeleteLiveHash} transaction,
+     * returning the metadata required to, at minimum, validate the signatures of all required
+     * signing keys.
+     *
+     * @param txn a transaction with a {@link
+     *     com.hederahashgraph.api.proto.java.CryptoDeleteLiveHashTransactionBody}
+     * @return the metadata for the live hash deletion
+     */
+    TransactionMetadata preHandleDeleteLiveHash(TransactionBody txn);
+
+    // HAPI Token Service
+
+    /**
+     * Pre-handles a {@link com.hederahashgraph.api.proto.java.HederaFunctionality#TokenCreate}
+     * transaction, returning the metadata required to, at minimum, validate the signatures of all
+     * required signing keys.
+     *
+     * @param txn a transaction with a {@link
+     *     com.hederahashgraph.api.proto.java.TokenCreateTransactionBody}
+     * @return the metadata for the token creation
+     */
+    TransactionMetadata preHandleCreateToken(TransactionBody txn);
```

Comment:
I think we need `TokenPreTransactionHandler` , but it can be in `service.token` ?

Reply:
Extracted `TokenService` and `TokenPreTransactionHandler`.